### PR TITLE
use more compact integer headers

### PR DESCRIPTION
**Pull request by ekoby**
use more compact wire representation for uint16/32/64 headers

`uint32(1)` would translate to 1 byte instead of 4

### CODE REVIEW

**Review by michaelquigley — APPROVED** [review]
Seems reasonable to me. Will definitely introduce an incompatibility with previous versions.

File: channel2/message_test.go [hunk 1]
```diff
@@ -61,3 +61,126 @@ func Test_getRetryVersionFor(t *testing.T) {
 		})
 	}
 }
```
Comment (plorenz) [inline channel2/message_test.go:63]: Can you add a test and see if decode still works on values encoded with binary.LittleEndian?

Comment (ekoby) [reply channel2/message_test.go:64]: added as part of Uint64 tests

**Comment by plorenz** [comment]
channel code has been moved to a new project, so this is outdated. The branch was ported over as well: https://github.com/openziti/channel/tree/compact-int-headers in case we want to revive this

**Pull request closed without merging**